cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -177,6 +157,15 @@ type ImageMeta struct {
 	//TODO: evaluate if we need also a ImageName based on user feedbacks
 }
 
+// AddOn defines settings to be used when deploying a specific addon kind
+type AddOn struct {
+	// Kind defines the addon type
+	Kind string
+
+	// ImageMeta allows to customize the image used for the addon
+	ImageMeta
```

Comment:
just wanting to get some discussion on how we are going to handle upgrades for these.
until now:
- kube-proxy versions is the k8s version
- coredns version is pinned in kubeadm

if the user feeds custom values, on upgrade do we:
- override the user tags with the sane defaults from the latest kubeadm?
- if they are modified tags/repo leave them as is and leave it to the user to patch the deployments and the kubeadm CM post upgrade? we just merged a coredns PR for the migration to skip if the tag or CM is unknown.

a side question, kube-proxy support X-1 skew? e.g. k8s at 1.17, but kube-proxy at 1.16.

Reply:
kube-proxy support the skew IIRC. Although, we might want to keep it in sync with kubelet since they both share the  `KUBE-MARK-DROP` and `KUBE-MARK-MASQ` iptable chains

Reply:
Right now `ImageMeta` in the kube-proxy case is completely ignored. No behavior change is introduced in this kubeadm version. `ImageMeta` is acknowledged only for CoreDNS and kube-dns.

Reply:
> ImageMeta is acknowledged only for CoreDNS and kube-dns.

ok, this goes to the question from my comment above:

> if the user feeds custom values, on upgrade via ImageMeta do we: